### PR TITLE
Oclif migration

### PR DESCRIPTION
**Pull request by mikesimos**
##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -4,4 +4,16 @@ root = true
...
 indent_style = tab
 end_of_line = lf
-charset = utf-8
\ No newline at end of file
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.json]
+indent_style = space
```
Comment (hedss) [inline .editorconfig]: This should be tabs: https://github.com/balena-io/process/blob/master/process/TypeScript_Coding_Guide.md#indentation and not spaces. What was followed to add this instead?

File: README.md [hunk 2]
```diff
@@ -1,132 +1,89 @@
...
-- `config-manifest.yml`: Specifies the schema of the environment configuration.
-- `templates/`:  Folder containing deployment spec templates in
-[mustache](https://mustache.github.io/) format, bearing extension `.tpl.ext`
-(ext may be `.yml` for example for docker-compose or kubernetes targets).
-- `environment.yml`: A file declaring available environments.
-
+## `katapult deploy [FILE]`
 
-###### `environment.yml`
+describe the command here
```
Comment (hedss) [inline README.md]: Probably worth describing the command here... ;-)

File: README.md [hunk 3]
```diff
@@ -1,132 +1,89 @@
...
 
-`--service-format`: Service format for a component as: --service-format
-`<component>=<format>`. Format could be either 'image' (component will use a
-pre-built Docker image) or 'build' (component will build the Docker image from
-local sources). Defaults to image for all components.
+## `katapult help [COMMAND]`
 
-`--build-path`: build path for a component as: --build-path
-`<component>=<path>`. Defaults to `./component-name`
+display help for katapult
```
Comment (hedss) [inline README.md]:
```suggestion
Display help for katapult
```

File: src/commands/deploy.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { Command } from '@oclif/command'
+
+import generateDeploy = require('../lib/commands/generate-deploy')
+
+import { GenerateDeployFlags } from './generate'
+
+export default class Deploy extends Command {
+  static description =
```
Comment (hedss) [inline src/commands/deploy.ts]: Tabs instead of 2 spaces.

File: src/commands/generate.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
+import { Command, flags } from '@oclif/command'
+
+import generateDeploy = require('../lib/commands/generate-deploy')
+
+export const GenerateDeployFlags = {
+  configuration: flags.string({
```
Comment (hedss) [inline src/commands/generate.ts]: Tabs instead of 2 spaces.

File: src/commands/deploy.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    'Generate Deploy Spec from environment configuration and deploy'
+
+  static flags = GenerateDeployFlags
+
+  static args = []
+
+  async run() {
+    const { flags } = this.parse(Deploy)
```
Comment (hedss) [inline src/commands/deploy.ts]: We use semi-colons at end of lines: https://github.com/balena-io/process/blob/master/process/TypeScript_Coding_Guide.md#semicolons

File: src/commands/generate.ts [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  }),
+
+  deploy: flags.boolean({
+    hidden: true,
+    default: false
+  })
+}
+
+export default class Generate extends Command {
+  static description = 'Generate Deploy Spec from environment configuration'
```
Comment (hedss) [inline src/commands/generate.ts]: Semi-colons at end of statements: https://github.com/balena-io/process/blob/master/process/TypeScript_Coding_Guide.md#semicolons

File: src/lib/utils.js [hunk 8]
```diff
@@ -6,7 +6,7 @@ const {
 	readFileAsync,
 	writeFileAsync,
 	statAsync,
-	renameAsync,
+	renameAsync
```
Comment (hedss) [inline src/lib/utils.js]: Trailing commas: https://github.com/balena-io/process/blob/master/process/TypeScript_Coding_Guide.md#trailing-commas
```suggestion
	renameAsync,
```

File: .editorconfig (hunk 1 above)
Comment (mikesimos) [reply .editorconfig]: Heya @hedss, this rule targets only *.json files, not typescript. (This aligns with json indentation across all components). Should we switch package.json and package.lock.json to tabs indentation (maybe everywhere)?

Comment (hedss) [reply .editorconfig]: @mikesimos Sorry, yes, you're absolutely right, and it should stick as it is!

Comment (mikesimos) [reply .editorconfig]: No worries man! And thanks for checking this :)

**Comment by mikesimos** [comment]
Hey @hedss, grand thanks for the review dude! Addressed!

**Comment by mikesimos** [comment]
Once again, thanks for pointing that out @hedss !

**Comment by hedss** [comment]
@mikesimos No worries, glad to help. Thanks for making the changes!

**Comment by mikesimos** [comment]
@hedss Thanks for the (always) great suggestions :)

**Pull request merged**